api/app/src/main/kotlin/packit/service/UserRoleFilterService.kt

```diff
@@ -0,0 +1,153 @@
+package packit.service
+
+import org.springframework.stereotype.Service
+import packit.model.Packet
+import packit.model.Role
+import packit.model.User
+import packit.model.dto.RolesAndUsers
+import packit.security.PermissionChecker
+
+interface UserRoleFilterService
+{
+    fun getRolesAndUsersWithSpecificReadPacketGroupPermission(
+        roles: List<Role>,
+        users: List<User>,
+        packetGroupName: String
+    ): RolesAndUsers
+
+    fun getRolesUsersWithSpecificReadPacketPermission(
```

Comment:
```suggestion
    fun getRolesAndUsersWithSpecificReadPacketPermission(
```

This matches the other function names better.